apps/web/docker/Dockerfile

```diff
@@ -1,25 +1,63 @@
-FROM latitudedata/base AS installer
-
-ARG NODE_ENV=development
+FROM latitudedata/base AS base
 
 RUN turbo prune @latitude-data/web --docker
 
-FROM installer AS runner
+FROM base AS prunner
 
-COPY .gitignore .gitignore
-COPY --from=installer /app/out/full .
+# COMMON stage (installer)
+# ------------------------------------------------------
+FROM base AS installer
+COPY --from=prunner /app/out/json/ .
+COPY --from=prunner /app/out/pnpm-lock.yaml ./pnpm-lock.yaml
+COPY --from=prunner /app/out/pnpm-workspace.yaml ./pnpm-workspace.yaml
+COPY --from=prunner /app/out/full/ .
 
-RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
-    if [ "$NODE_ENV" = "production" ]; then \
-        pnpm install --prod --frozen-lockfile; \
-    else \
-        pnpm install; \
-    fi
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install \
+  --frozen-lockfile \
+  --filter "@latitude-data/web..."
 
+# DEVELOPMENT stage
+FROM installer AS dev-runner
 WORKDIR /app/apps/web
+ENV NODE_ENV=development
+CMD pnpm dev
+
+# PRODUCTION stager
+# ------------------------------------------------------
+FROM installer AS prod-deps
+WORKDIR /app
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install \
+  --prod \
+  --frozen-lockfile
```

Comment:
another install?

Reply:
They're different stages. But let's talk offline. Not sure if it's the best approach.